### PR TITLE
Fix spiffetls dial Segfault

### PR DESCRIPTION
**Pull request by willallves**

### CODE REVIEW

File: spiffetls/src/c_dial_example.c [hunk 1]
```diff
@@ -1,8 +1,16 @@
 #include "spiffetls/src/dial.h"
 #include <unistd.h>
 
+void init_openssl()
+{
+    SSL_load_error_strings();
+    //OpenSSL_add_ssl_algorithms();
```
Comment (otaviolcs3) [inline spiffetls/src/c_dial_example.c:7]:
```suggestion
```

File: spiffetls/src/dial.c [hunk 2]
```diff
@@ -64,23 +69,27 @@ SSL *spiffetls_DialWithMode(in_port_t port, in_addr_t addr,
...
         goto error;
     }
-
-    SSL *conn = SSL_new(tls_config);
-
     const int sockfd
         = config->dialer_fd > 0 ? config->dialer_fd : createSocket(addr, port);
 
     if(sockfd < 0) {
-        // could not create socket with given address and port
```
Comment (otaviolcs3) [inline spiffetls/src/dial.c:74]: Insert comment back.

File: spiffetls/src/dial.c [hunk 3]
```diff
@@ -64,23 +69,27 @@ SSL *spiffetls_DialWithMode(in_port_t port, in_addr_t addr,
...
+
+    if(!conn) {
+        goto error;
+    }
+
+    if(SSL_set_fd(conn, sockfd) != 1) {
+        goto error;
+    }
 
     SSL_set_fd(conn, sockfd);
```
Comment (otaviolcs3) [inline spiffetls/src/dial.c:89]: SSL_set_fd is being call twice.

File: spiffetls/src/dial.c [hunk 4]
```diff
@@ -64,23 +69,27 @@ SSL *spiffetls_DialWithMode(in_port_t port, in_addr_t addr,
...
     SSL_set_connect_state(conn);
 
     if(SSL_connect(conn) != 1) {
-        // could not build a SSL session
```
Comment (otaviolcs3) [inline spiffetls/src/dial.c:83]: Insert comment back.

**Review by otaviolcs3 — APPROVED** [review]

File: docker/grpc.Dockerfile [hunk 5]
```diff
@@ -40,11 +43,19 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         checkinstall \
         zlib1g-dev \
         doxygen \
-        graphviz
+        graphviz \
+        docker-compose
```
Comment (rodrigolc) [inline docker/grpc.Dockerfile:47]: docker compose inside the container?

File: docker/grpc.Dockerfile [hunk 6]
```diff
@@ -40,11 +43,19 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
...
 
 RUN pip3 install behave PyHamcrest pathlib2
 RUN apt-get clean
 
+# Install Docker Enginer
```
Comment (rodrigolc) [inline docker/grpc.Dockerfile:52]: why install docker in the container?

Comment (rodrigolc) [reply docker/grpc.Dockerfile:52]: Got it, but shouldn't those be installed on another image that extends the build image?

**Review by rodrigolc — CHANGES_REQUESTED** [review]

File: spiffetls/src/c_dial_example.c [hunk 7]
```diff
@@ -38,11 +45,25 @@ int main(void)
...
+
     if(err != NO_ERROR) {
-        printf("could not create TLS connection!");
+        printf("could not create TLS connection\n");
         exit(-1);
     }
 
+    char buff[1024];
+    int bytes;
+    char *message = "Hi Server!";
```
Comment (arianalima) [inline spiffetls/src/c_dial_example.c:60]: It would be nice if the message could be passed as an argument when running the example

Comment (willallves) [reply spiffetls/src/c_dial_example.c]: done

**Review by rodrigolc — APPROVED** [review]

**Pull request merged**